src/goto-instrument/nondet_static.cpp

```diff
@@ -50,7 +51,7 @@ void nondet_static(
         continue;
 
       // constant?
-      if(sym.type().get_bool(ID_C_constant))
+      if(is_constant_variable_or_array(sym.type()))
         continue;
```

Comment:
I don't think this is quite the right logic because there are a number of cases that it doesn't cover.  What about structs?  What about unions?  What about the difference between const int * and int * const?

[ I always mix the last ones up so I use ```cdelc```:
```
cdecl> explain const int *s
declare s as pointer to const int
cdecl> explain const int * const s
declare s as const pointer to const int
cdecl> explain int * const s
declare s as const pointer to int
```
]